docs/blog/2019-07-10-user-testing-accessible-client-routing/index.md

```diff
@@ -0,0 +1,181 @@
+
+With this data, we can make better recommendations for accessible client-rendered websites in general and at the JavaScript framework level. By adjusting Gatsby’s implementation of client-side routing–currently leveraging [@reach/router](https://github.com/reach/router) for React.js–to better support a range of people with disabilities, we can improve access for users and also potentially influence accessible UI patterns in other JavaScript frameworks. Eventually (we hope), this work could encourage new browser APIs through web standards: “paving the accessible cowpaths” with solutions based in user research.
+
+## What is client-side routing?
+
+In client-rendered JavaScript applications–also referred to as Single Page Apps–traditional HTML page reloads don’t typically occur when a user navigates through the app. Instead, client-side JavaScript handles routing through the app by controlling the browser’s history and mapping client-rendered URLs to each page or view. To ensure these experiences are accessible, developers have to recreate some of the missing browser feedback for users by manually managing focus and making announcements in assistive technology (AT).
+
+## Client-rendering in Gatsby
+
+Gatsby [builds](/docs/glossary#build) [static]/docs/glossary#static) HTML pages with Node.js (a.k.a. “SSR”), which is great for both accessibility and performance: by default, pages will render without JavaScript and links will navigate through the site with traditional page refreshes like they should. With client-side JavaScript enabled, Gatsby scripts will download and [hydrate](/docs/glossary#hydration) the site into a full React.js web application that can manipulate the [DOM](/docs/glossary#dom) and provide a smooth user experience. In this context, Gatsby sites must be accessible and use the best techniques possible, hence the desire for user research data to better support people with disabilities.
+
+## Common accessibility barriers in client-rendered web apps
+
+These are some common accessibility barriers when it comes to client-side routing:
+
+- Without page refreshes, screen reader users may not be informed that the page has changed.
+- Without a page refresh or focus management, a user’s keyboard focus point may be kept in the same place as where they clicked, which isn’t intuitive.
+  - In layouts where the page changes partially to include a deep-linked modal dialog or other view layer, a user’s focus point could be left in an entirely wrong spot on the page.
+- Without visible focus outlines, sighted keyboard, switch, and voice dictation users can’t see their focus point on the screen when views change with focus management.
+
+Some of the commonly recommended solutions include (references at the end of this post):
+
+- Dynamically set focus to an HTML wrapper element on page change, to both move focus to the new content and make an announcement in AT.
```

Comment:
"AT" acronym is explained/defined in earlier section, but readers might need to look it up again if they are not very familiar with the topic. It might make article more accessible (sorry, not sorry) if full "assistive technology" term was used instead of "AT".

Not strong opinion on this - I just remembered I had to look it up few times in the past when I was reading other articles.

Reply:
I think I would second @pieh's suggestion. I read the blog post in a few small bursts on my phone and had forgotten what AT was short for at one point. :)

Reply:
Good feedback, thank you!